File: herramienta1-selenium/src/test/java/com/example/steps/LoginSteps.java
package com.example.steps;

import com.example.pages.LoginPage;
import com.example.hooks.Hooks;

import io.cucumber.java.en.Given;
import io.cucumber.java.en.When;
import io.cucumber.java.en.Then;

import org.openqa.selenium.WebDriver;
import org.junit.Assert;

public class LoginSteps {

    WebDriver driver;
    LoginPage loginPage;

    @Given("el usuario navega a la página principal")
    public void navegarPaginaPrincipal() {

        driver = Hooks.driver;
        loginPage = new LoginPage(driver);
    }

    @When("inicia sesión con usuario {string} y password {string}")
    public void iniciarSesionConCredenciales(String usuario, String password) {

        loginPage.abrirModal();
        loginPage.completarLogin(usuario, password);
        loginPage.clickLogin();
    }

    @Then("el nombre del usuario aparece en la barra superior")
    public void validarLoginCorrecto() {

        String mensaje = loginPage.obtenerUsuarioLogueado();

        Assert.assertTrue(
                mensaje.contains("UsuarioExiste")
        );
    }
}

File: herramienta2-playwright/steps/loginSteps.js
const { Given, When, Then } = require('@cucumber/cucumber');

Given('el usuario está en la página principal', async function () {
    await this.homePage.goTo();
});

When('inicia sesión con usuario {string} y password {string}', async function (username, password) {
    await this.homePage.openLoginModal();
    await this.loginPage.login(username, password);
});

Then('el nombre del usuario aparece en la barra superior', async function () {
    const texto = await this.homePage.getUsername();
    if (!texto.includes('Welcome UsuarioExiste')) throw new Error('Usuario no autenticado');
});